src/DIPS.Xamarin.UI/Controls/Toast/ToastCore.cs

```diff
@@ -109,20 +84,28 @@ private async Task HideToast(ContentPage currentPage, bool removeContainer)
             {
                 // create and register toast container
                 toastContainer = new Grid();
-                RegisterName(currentPage.Id.ToString(), toastContainer);
+                RegisterName(m_currentPageWithToast.Id.ToString(), toastContainer);
 
                 // old content
-                var oldContent = currentPage.Content;
+                var oldContent = m_currentPageWithToast.Content;
 
                 // set new content
-                await MainThread.InvokeOnMainThreadAsync(() => { currentPage.Content = toastContainer; });
```

Comment:
Yes, invoking on `MainThread` can be removed and let the user wrap it if it's needed, just like in other `Xamarin.Forms` controls. (Needed in cases where the Toast is displayed from a ViewModel after a Background service thread invokes it)